django_netjsonconfig/base/template.py

```diff
@@ -100,5 +103,21 @@ def clean(self, *args, **kwargs):
         if self.type == 'vpn' and not self.config:
             self.config = self.vpn.auto_client(auto_cert=self.auto_cert)
 
+    def clone(self, user):
+        clone = copy(self)
```

Comment:
There's an official way to clone model instances, please see https://docs.djangoproject.com/en/2.2/topics/db/queries/#copying-model-instances.

Reply:
I've read this, but this official way requires modification of the original object instance, IMO it would be quite confusing if a function named 'clone' behaved that way, what do you think?

Reply:
I am not sure how the is the orginal one affected in the official way? 
I may be missing something, can you please explain that, thanks! :smile:

Reply:
The official way is to just change the primary key to None, do some changes and save() the original object instance. Change of the pk makes it a new object in database and when you are operating directly on the objects, it's good approach but lets see what happened if I implemented the helper clone() function this way:
Let's have some `original` Template
`clone = original.clone()`
The clone is created, but now, the `original` instance is equal to `clone`.
In the clone action implementation it doesn't make a difference since it does not longer use the `original` object, but do You not think that behavior may be confusing a bit if one wanted to reuse that clone function?

Reply:
Ah! I took me a really long while, been reading your message for 15 mins but I think now I get it! :smile:
The `self` object would be changed! 
CC: @nemesisdesign

Reply:
Sorry for being incomprehensible,
If you really want to do it similarly to the official way, It's also possible to change `copy(self)` to
`self.__class__.objects.get(pk=self.pk)`, but IMO the copy() function is a way more readable.